tutorials/preprocessing/plot_55_setting_eeg_reference.py

```diff
@@ -0,0 +1,122 @@
+# -*- coding: utf-8 -*-
+"""
+.. _tut-set-eeg-ref:
+
+Setting the EEG reference
+=========================
+
+This tutorial describes how to set an EEG reference in MNE-Python.
+
+.. contents:: Page contents
+   :local:
+   :depth: 2
+
+As usual we'll start by importing the modules we need, loading some
+:ref:`example data <sample-dataset>`, and cropping it to save memory:
+"""
+
+import os
+import mne
+
+sample_data_folder = mne.datasets.sample.data_path()
+sample_data_raw_file = os.path.join(sample_data_folder, 'MEG', 'sample',
+                                    'sample_audvis_raw.fif')
+raw = mne.io.read_raw_fif(sample_data_raw_file, verbose=False)
+raw.crop(tmax=60).load_data()
+
+###############################################################################
+# Background
+# ^^^^^^^^^^
+#
+# Typically, one of the first steps in processing EEG recordings is to subtract
+# a *reference signal* from each channel. Conceptually, the reference signal
+# represents environmental or equipment noise that affects all sensors
+# approximately equally, and by subtracting it from the signal at each
+# electrode, what is left will be a less noisy representation of brain activity
+# than the original raw signal.
+#
+# Sometimes the subtracted reference signal is the signal from a physical
+# electrode (typical reference electrode placements are the earlobe or the
+# mastoid processes) or the average from a pair of such electrodes. Other
+# times, the subtracted reference signal is the average of signals at all
+# electrodes. MNE-Python supports all of these possibilities through the
+# :meth:`~mne.io.Raw.set_eeg_reference` method. Of course, you can only set
+# the reference to a specific (set of) electrode(s) if those electrodes are
+# present in the :class:`~mne.io.Raw` file; the :ref:`example data
+# <sample-dataset>` has numeric EEG channel names (``EEG 001``, etc) so the
+# double mastoid and earlobe examples are commented out here:
```

Comment:
I'd explicitly mention `mne.add_reference_channels` here to show how new reference channels that are not yet part of the data set can be added.

Reply:
I didn't know about that function. I'm not sure I fully understand how one would use it in normal circumstances... @cbrnr can you enlighten me?  Other than adding an all-zero channel, how is it functionally different from `set_eeg_reference([])`?

Reply:
Like you said, `add_reference_channels` adds one (or more) reference channels to the data array. These are by definition all zero. `set_eeg_reference` can't add channels back, but when re-referencing this is sometimes needed. Assume your current reference is Cz and this all-zero channel is not in the data. Now you want to compute the average reference, but you want a signal at Cz - therefore, you have to add the reference Cz back to the data before doing the average reference.